example/examples/echo-server.rs

```diff
@@ -67,10 +80,13 @@ fn main() {
     // incoming connections, and that will automatically apply all the selected protocols on top
     // of any opened stream.
 
-    // We use it to listen on `/ip4/127.0.0.1/tcp/10333`.
-    let future = transport.listen_on(swarm::Multiaddr::new("/ip4/0.0.0.0/tcp/10333").unwrap())
-        .unwrap_or_else(|_| panic!("unsupported multiaddr protocol ; should never happen")).0
+    // We use it to listen on the address.
+    let (listener, address) = transport
+        .listen_on(swarm::Multiaddr::new(&listen_addr).expect("invalid multiaddr"))
+        .unwrap_or_else(|_| panic!("unsupported multiaddr protocol ; should never happen"));
```

Comment:
Assumably this is because `listen_on`'s error type doesn't implement `Debug`, right? Maybe use `map_err(...).expect(...)` with a comment above the `map_err` saying why we can't use `expect`.

Reply:
And another point is that now that the address is no longer hardcoded, this panic can actually happen.

Reply:
Maybe we can do `.map_err(|(_, addr)| addr).expect("Unsupported multiaddr")`